tools/wptrunner/wptrunner/testrunner.py

```diff
@@ -559,7 +561,19 @@ def run_test(self):
             self.logger.info("Run %d/%d" % (self.run_count, self.rerun))
             self.send_message("reset")
         self.run_count += 1
+        wait_timeout = (self.state.test.timeout * self.executor_kwargs['timeout_multiplier'] +
+                        3 * self.executor_cls.extra_timeout)
+        self.timer = threading.Timer(wait_timeout, self._timeout)
         self.send_message("run_test", self.state.test)
+        self.timer.start()
+
+    def _timeout(self):
+        self.logger.info("Got timeout in harness")
+        test = self.state.test
+        self.test_ended(test,
+                        (test.result_cls("EXTERNAL-TIMEOUT",
+                                         "Executor hit external timeout "
```

Comment:
Shall we have a different message here to differentiate this from the `EXTERNAL-TIMEOUT` from executors? Maybe "TestRunner hit external timeout"?